src/refactor_bot/agents/refactor_executor.py

```
@@ -109,19 +234,60 @@ def execute(
-            raise ExecutionError(f"Failed to call Claude API: {e}") from e
+        # Step 5: Call LLM API
+        response = None
+        used_provider = ""
+        providers = self._provider_chain()
+        last_error: Exception | None = None
+        tool_schema = self._get_tool_schema()
+        for index, provider in enumerate(providers):
+            try:
+                if provider == "anthropic":
+                    if not self._anthropic_client:
+                        raise ExecutionError("Anthropic client unavailable")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)